src/extension.js

```diff
@@ -93,7 +93,11 @@ const ServiceIndicator = GObject.registerClass({
         this._item.label.clutter_text.x_expand = true;
         this.menu.addMenuItem(this._item);
 
-        AggregateMenu.menu.addMenuItem(this.menu, 4);
+        // Find current index of network menu
+        let menuItems = AggregateMenu.menu._getMenuItems();
+        let networkMenuIndex = menuItems.indexOf(AggregateMenu._network.menu) || 3;
```

Comment:
It's probably a rare case, but if the item is not in the menu `indexOf()` will return `-1` which is actually a truthy value so the expression will never evaluate to `3`.

Maybe remove the `||` and add a second line:

```js
let menuIndex = networkMenuIndex > -1 ? networkMenuIndex : 3;
```

Reply:
Oof, good catch, thanks!

Reply:
Should be resolved by e71a54e

Reply:
LGTM, thanks!